src/Database/Adapter/MysqlAdapter.php

```diff
@@ -18,13 +22,146 @@ public function getQueryBuilder()
     }
 
-    public function tableInfo($table)
+    protected function loadStructure()
     {
-        throw new LogicException('Not yet implemented');
+        $database = $this->execute('SELECT database()')->fetchColumn();
+        $structure = new Structure();
+        $tables = $this->execute("SELECT * FROM information_schema.TABLES WHERE TABLE_SCHEMA = '$database' ORDER BY TABLE_NAME")->fetchAll(PDO::FETCH_ASSOC);
+        foreach ($tables as $table) {
+            $migrationTable = $this->tableInfo($table['TABLE_NAME'], $database);
+            if ($migrationTable) {
+                $structure->update($migrationTable);
+            }
+        }
+        return $structure;
+    }
+
+    private function tableInfo($table, $database)
+    {
+        $migrationTable = new MigrationTable($table);
+        $this->loadColumns($migrationTable, $table);
+        $this->loadIndexes($migrationTable, $table);
+        $this->loadForeignKeys($migrationTable, $database, $table);
+        return $migrationTable;
     }
 
     protected function createRealValue($value)
     {
         return is_array($value) ? implode(',', $value) : $value;
     }
+
+    protected function escapeString($string)
+    {
+        return '`' . $string . '`';
+    }
+
+    private function remapType($type)
+    {
+        $types = [
+            'int' => Column::TYPE_INTEGER,
+            'tinyint' => Column::TYPE_TINY_INTEGER,
+            'smallint' => Column::TYPE_SMALL_INTEGER,
+            'mediumint' => Column::TYPE_MEDIUM_INTEGER,
+            'bigint' => Column::TYPE_BIG_INTEGER,
+            'varchar' => Column::TYPE_STRING,
+            'linestring' => Column::TYPE_LINE,
+        ];
+        return isset($types[$type]) ? $types[$type] : $type;
+    }
+
+    private function loadColumns(MigrationTable $migrationTable, $table)
+    {
+        $columns = $this->execute(sprintf('SHOW FULL COLUMNS FROM `%s`', $table))->fetchAll(PDO::FETCH_ASSOC);
+        foreach ($columns as $column) {
+            $values = null;
+            $type = $column['Type'];
+            preg_match('/(.*?)\((.*?)\)(.*)/', $column['Type'], $matches);
+            if (isset($matches[1]) && $matches[1] != '') {
+                $type = $matches[1];
+            }
+
+            $type = $this->remapType($type);
+
+            if (($type == 'enum' || $type == 'set') && isset($matches[2])) {
+                $values = explode('\',\'', substr($matches[2], 1, -1));
+            }
+
+            $length = null;
+            $decimals = null;
+            if (isset($matches[2])) {
+                if (strpos($matches[2], ',')) {
+                    list($length, $decimals) = explode(',', $matches[2], 2);
+                    $length = (int) $length;
+                    $decimals = (int) $decimals;
+                } else {
+                    $length = (int) $matches[2];
+                }
+            }
+
+            if ($type == Column::TYPE_CHAR && $length == 36) {
```

Comment:
Maybe add a configuration for this. Currently I can't find a situation where to use an size of exactly 36 chars vor a field but anyways this implies not to be an UUID IMHO